connection.go

```diff
@@ -28,19 +28,22 @@ type mysqlContext interface {
 }
 
 type mysqlConn struct {
-	buf              buffer
-	netConn          net.Conn
-	affectedRows     uint64
-	insertId         uint64
-	cfg              *Config
-	maxAllowedPacket int
-	maxWriteSize     int
-	writeTimeout     time.Duration
-	flags            clientFlag
-	status           statusFlag
-	sequence         uint8
-	parseTime        bool
-	strict           bool
+	buf                 buffer
+	netConn             net.Conn
+	affectedRows        uint64
+	insertId            uint64
+	cfg                 *Config
+	maxAllowedPacket    int
+	maxWriteSize        int
+	writeTimeout        time.Duration
+	flags               clientFlag
+	status              statusFlag
+	sequence            uint8
+	compressionSequence uint8
+	parseTime           bool
+	strict              bool
+	reader              packetReader
+	writer              io.Writer
```

Comment:
Do we really need to carry around both `reader` and `writer` as well as `buf` and `netConn`?
Can't those be unified into one interface?

This interface might also have some sort of sequence reset function. Then `sequence` and `compressionSequence` could be internal states of the respective struct implementing the interface.